File: java/src/main/java/gherkin/ast/TableRow.java
package gherkin.ast;

import java.util.Collections;
import java.util.List;

public class TableRow extends Node {
    private final List<TableCell> cells;

    public TableRow(Location location, List<TableCell> cells) {
        super(location);
        this.cells = Collections.unmodifiableList(cells);
    }

    public List<TableCell> getCells() {
        return cells;
    }

    @Override
    public void accept(Visitor visitor) {
        for (TableCell cell : cells) {
            cell.accept(visitor);
        }
        visitor.visitTableRow(this);
    }
}

File: csharp/Gherkin/Ast/TableRow.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Gherkin.Ast
{
    public class TableRow : IHasLocation
    {
        public Location Location { get; private set; }
        public IEnumerable<TableCell> Cells { get; private set; }

        public TableRow(Location location, TableCell[] cells)
        {
            Location = location;
            Cells = cells;
        }
    }
}